src/structures/paging/frame.rs

```diff
@@ -148,6 +148,18 @@ impl<S: PageSize> PhysFrameRange<S> {
     pub fn is_empty(&self) -> bool {
         self.start >= self.end
     }
+
+    /// Returns the number of frames in the range.
+    #[inline]
+    pub fn len(&self) -> u64 {
+        (self.end.start_address() - self.start.start_address()) / S::SIZE
+    }
+
+    /// Returns the size in bytes of all frames within the range.
+    #[inline]
+    pub fn size(&self) -> u64 {
+        self.start.size() * self.len()
```

Comment:
```suggestion
        self.end.start_address() - self.start.start_address()
```

Let's avoid multiplications if we can.

Reply:
The compiler should be smart enough to remove the multiplication after inlineing.

I mean I can change this, but with the panic you pointed out above this would become quite ugly, especially for the range-inclusive version.

Reply:
Fair enough.